options.go

```diff
@@ -2,40 +2,27 @@ package discovery
 
 import (
 	"time"
-)
 
-// Option is a single discovery option
-type Option func(opts *Options) error
+	moved "github.com/libp2p/go-libp2p-core/discovery"
+)
 
-// Options is a set of discovery options
-type Options struct {
-	Ttl   time.Duration
-	Limit int
-	// Other (implementation-specific) options
-	Other map[interface{}]interface{}
-}
+// Deprecated: use skel.DiscoveryOpt instead.
+type Option = moved.DiscoveryOpt
 
-// Apply applies the given options to this Options
-func (opts *Options) Apply(options ...Option) error {
-	for _, o := range options {
-		if err := o(opts); err != nil {
-			return err
-		}
-	}
-	return nil
-}
+// Deprecated: use skel.DiscoveryOpts instead.
+type Options = moved.DiscoveryOpts
 
 // TTL is an option that provides a hint for the duration of an advertisement
-func TTL(ttl time.Duration) Option {
-	return func(opts *Options) error {
+func TTL(ttl time.Duration) moved.DiscoveryOpt {
```

Comment:
shouldn't the options move too?

Reply:
SGTM; they are not implementation-specific.

What about the util methods under `util.go`? They are implementation-agnostic.

Reply:
I think the whole package can be moved actually. It's all utility functions with no dependency on concrete implementations.

Reply:
Actually let's keep `routing` here -- we will eventually add rendezvous too.
But the utilities can certainly move.

Reply:
Hm, but routing doesn't deal with implementation-specific details, it just deals with abstractions. In fact, go-libp2p-rendezvous should just satisfy the Discovery interface. I think this repo should go away.